internal/metadataextractor/extractor.go

```diff
@@ -401,26 +418,24 @@ func (e *ExtractorHandler) ExtractMetadata(ctx context.Context, method_name stri
 	}
 	ctx, cancel := context.WithTimeout(ctx, e.timeout)
 	defer cancel()
-	if err := runExtractor(ctx, binary_path, args, stdout_callback, stderr_callback); err == nil {
 
-		if ctx.Err() == context.DeadlineExceeded {
-			slog.Error("Extraction timed out")
-			return "", ctx.Err()
-		}
+	err = runExtractor(ctx, binary_path, args, stdout_callback, stderr_callback)
```

Comment:
Question: should we also log errors instead of just returning them to the caller? It seems beneficial to have a history of errors in the log of the service. That's why I also put the `slog.Error(...)` statements there
Thoughts?

Reply:
We could make ticket to go through all the logging and align; use slog. everywhere, config log file, log all errors,... we can agree on the exact content